### PR TITLE
fix: use digest when loading image unto daemon

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

File: oci/private/structure_test.bzl [hunk 1]
```diff
@@ -24,9 +24,12 @@ exec "{st_path}" test {fixed_args} "$@"
 def _structure_test_impl(ctx):
     st_info = ctx.toolchains["@contrib_rules_oci//oci:st_toolchain_type"].st_info
 
-    default_image_tag = "{workspace}.local/{package}/{name}:latest".format(
-        workspace = ctx.workspace_name,
-        package = ctx.label.package.replace("/", "_"),
+    workspace = "default_workspace" if ctx.workspace_name == "__main__" else ctx.workspace_name
```
Comment (alexeagle) [inline oci/private/structure_test.bzl:27]: Instead of that special case how about just escaping whatever name? __foo__ is a legal workspace name

Comment (thesayyn) [reply oci/private/structure_test.bzl:27]: right... that'd be better. I'll fix it.

Comment (thesayyn) [reply oci/private/structure_test.bzl]: done

File: oci/private/structure_test.bzl [hunk 2]
```diff
@@ -21,12 +21,20 @@ CMD = """\
 exec "{st_path}" test {fixed_args} "$@"
 """
 
+def _strip_illegal_chars(str):
```
Comment (alexeagle) [inline oci/private/structure_test.bzl:24]: "Strip" suggests throw away, how about escape?

Also I would just replace with one character

File: oci/private/structure_test.bzl [hunk 3]
```diff
@@ -21,12 +21,20 @@ CMD = """\
...
+    return str.replace("_", "underscore").replace("/", "slash")
```
Comment (alexeagle) [inline oci/private/structure_test.bzl:25]: Are those the only illegal chars?

**Review by alexeagle — COMMENTED** [review]
actually I think this is a regression.

previously you could write some user instructions like
`bazel build :some-image; podman run bazel/some-image:latest`
but now you can't predict it anymore

**Comment by thesayyn** [comment]
we have `oci_tarball` for that and people should use that since images guaranteed to have tags you specify via `repotags` attribute.

**Review by alexeagle — APPROVED** [review]

**Pull request merged**